### PR TITLE
Fix non-functional tutorial in documentation

### PR DESCRIPTION
**Pull request by luciamunozdev**
**Description**:
This PR updates the documentation to reflect recent security model changes and introduces the `TokenAssociateTransaction` and `AccountAllowanceApproveTransaction` functions, which are now essential due to limitations in associating HTS tokens within smart contracts.

**Related issue(s)**:

Fixes #

**Notes for reviewer**:
1. **INVALID_FULL_PREFIX_SIGNATURE_FOR_PRECOMPILE**: This error occurred when attempting to use the `associateToken` function. It's essential to understand that due to the recent security model changes, this function can no longer be used for associating tokens, particularly those created with the Hedera Token Service (HTS). Developers should be aware of this limitation.

2. **SPENDER_DOES_NOT_HAVE_ALLOWANCE**: Another error, this one was observed in the context of token transfers. It's important to emphasize that, with the new security model in place, the smart contract must have necessary permissions from the token owner to authorize the transfer.

**Checklist**

- [X] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Comment by theekrystallee** [comment]
Hi @luciamunozdev, thanks so much for submitting this PR! We are coming across some errors/issues while testing the code. Could you send me the zip file with all the code, including your ENV file? My email is in my bio. Thanks!

**Comment by luciamunozdev** [comment]
Hi @theekrystallee ! I just sent you the email

File: tutorials/smart-contracts/deploy-a-contract-using-the-hedera-token-service.md [hunk 1]
```diff
@@ -437,18 +430,10 @@ System.out.println("The transaction status: " +associateTokenResponse.getReceipt
...
-      //The gas for the transaction
-      .setGas(2000000)
-      //The contract function to call and parameters to pass
-      .setFunction("tokenAssociate", new ContractFunctionParameters()
-           //The account ID to associate the token to
-           .addAddress(accountIdTest.toSolidityAddress())
-           //The token ID to associate to the account
-           .addAddress(tokenId.toSolidityAddress()));
+//Associate the token to an account using the SDK
+const associateToken = new TokenAssociateTransaction()
```
Comment (theekrystallee) [inline tutorials/smart-contracts/deploy-a-contract-using-the-hedera-token-service.md]: The code you emailed does not reflect this commit for JavaScript. Please update to:

```js
  const transaction = new TokenAssociateTransaction()
    .setAccountId(accountIdTest)
    .setTokenIds([tokenId])
    .freezeWith(client);
```

File: tutorials/smart-contracts/deploy-a-contract-using-the-hedera-token-service.md [hunk 2]
```diff
@@ -505,94 +505,86 @@ fmt.Printf("The associate transaction status %v\n", txStatus)
...
+
+	//Request the receipt of the transaction
+	const receipt = await txResponse.getReceipt(client);
 
-//The balance of the account
-const accountBalance = new AccountBalanceQuery()
-    .setAccountId(accountIdTest)
-    .execute(client);
+	//Get the transaction consensus status
+	const transactionStatus = receipt.status;
```
Comment (theekrystallee) [inline tutorials/smart-contracts/deploy-a-contract-using-the-hedera-token-service.md]:
```js
  //Get the transaction consensus status
  const transactionStatusAllowance = receiptAllowance.status;
```

File: tutorials/smart-contracts/deploy-a-contract-using-the-hedera-token-service.md [hunk 3]
```diff
@@ -505,94 +505,86 @@ fmt.Printf("The associate transaction status %v\n", txStatus)
...
+
+	console.log("The transaction consensus status for the allowance function is " + transactionStatus.toString());
```
Comment (theekrystallee) [inline tutorials/smart-contracts/deploy-a-contract-using-the-hedera-token-service.md]:
```js
  console.log(
    "The transaction consensus status for the allowance function is " +
      transactionStatusAllowance.toString()
  );
```

File: tutorials/smart-contracts/deploy-a-contract-using-the-hedera-token-service.md [hunk 4]
```diff
@@ -1044,22 +1026,14 @@ async function htsContract() {
...
-        //The gas for the transaction
-        .setGas(2000000)
-        //The contract function to call and parameters to pass
-        .setFunction("tokenAssociate", new ContractFunctionParameters()
-             //The account ID to associate the token to
-             .addAddress(accountIdTest.toSolidityAddress())
-             //The token ID to associate to the account
-             .addAddress(tokenId.toSolidityAddress()));
+    //Associate the token to an account using the SDK
+    const associateToken = new TokenAssociateTransaction()
```
Comment (theekrystallee) [inline tutorials/smart-contracts/deploy-a-contract-using-the-hedera-token-service.md]:
```js
  const transaction = new TokenAssociateTransaction()
    .setAccountId(accountIdTest)
    .setTokenIds([tokenId])
    .freezeWith(client);
```

File: tutorials/smart-contracts/deploy-a-contract-using-the-hedera-token-service.md [hunk 5]
```diff
@@ -1044,22 +1026,14 @@ async function htsContract() {
...
+        .setAccountId(accountIdTest)
+        .setTokenIds([tokenId])
 
     //Sign with the account key and submit to the Hedera network
     const signTx = await associateToken.freezeWith(client).sign(accountKeyTest);
-    
+
```
Comment (theekrystallee) [inline tutorials/smart-contracts/deploy-a-contract-using-the-hedera-token-service.md:1023]:
```js 
 const signTx = await transaction.sign(accountKeyTest);
```

File: tutorials/smart-contracts/deploy-a-contract-using-the-hedera-token-service.md [hunk 6]
```diff
@@ -1071,26 +1045,26 @@ async function htsContract() {
...
-    const childRecords = new TransactionRecordQuery()
-        //Set children equal to true for child records
-        .setIncludeChildren(true)
-        //The parent transaction ID
-        .setTransactionId(submitAssociateTx.transactionId)
-        .setQueryPayment(new Hbar(10))
-        .execute(client);
-    
+    //Approve the token allowance so that the contract can transfer tokens from the treasury account
+    const transaction = new AccountAllowanceApproveTransaction()
```
Comment (theekrystallee) [inline tutorials/smart-contracts/deploy-a-contract-using-the-hedera-token-service.md]:
```js
  const transactionAllowance = new AccountAllowanceApproveTransaction()
    .approveTokenAllowance(tokenId, treasuryAccountId, newContractId, 100)
    .freezeWith(client);
```

File: tutorials/smart-contracts/deploy-a-contract-using-the-hedera-token-service.md [hunk 7]
```diff
@@ -1071,26 +1045,26 @@ async function htsContract() {
...
+    const txResponse = await signTx.execute(client);
+
+    //Request the receipt of the transaction
+    const receipt = await txResponse.getReceipt(client);
+
+    //Get the transaction consensus status
+    const transactionStatus = receipt.status;
 
-    console.log("The " + tokenId + " should now be associated to my account: " + (await accountBalance).tokens.toString());
+    console.log("The transaction consensus status for the allowance function is " + transactionStatus.toString());
```
Comment (theekrystallee) [inline tutorials/smart-contracts/deploy-a-contract-using-the-hedera-token-service.md]:
```js
  //Sign the transaction with the owner account key
  const signTxAllowance = await transactionAllowance.sign(treasuryKey);

  //Sign the transaction with the client operator private key and submit to a Hedera network
  const txResponseAllowance = await signTxAllowance.execute(client);

  //Request the receipt of the transaction
  const receiptAllowance = await txResponseAllowance.getReceipt(client);

  //Get the transaction consensus status
  const transactionStatusAllowance = receiptAllowance.status;

  console.log(
    "The transaction consensus status for the allowance function is " +
      transactionStatusAllowance.toString()
  );
```

**Review by theekrystallee — CHANGES_REQUESTED** [review]

**Comment by theekrystallee** [comment]
Hi @luciamunozdev please resubmit after reviewing the comments and resolving the conflict. I believe we spoke about this over email as well.

**Comment by theekrystallee** [comment]
@luciamunozdev  I cannot get your code to run for all three languages. You can use this working code for JS as a reference for the other two languages and resubmit. Each code snippet you update should be reflected in the final "code check" code. There are also missing imports

```js
require("dotenv").config();

const {
  Hbar,
  Client,
  AccountId,
  TokenType,
  PrivateKey,
  AccountBalanceQuery,
  FileCreateTransaction,
  TokenCreateTransaction,
  ContractCreateTransaction,
  ContractExecuteTransaction,
  ContractFunctionParameters,
  AccountCreateTransaction,
  AccountAllowanceApproveTransaction,
  TokenAssociateTransaction,
} = require("@hashgraph/sdk");

// Import the compiled contract
const htsContract = require("./HTS.json");

async function htsContractFunction() {
  //Grab your Hedera testnet account ID and private key from your .env file
  const accountIdTest = AccountId.fromString(process.env.MY_ACCOUNT_ID);
  const accountKeyTest = PrivateKey.fromStringED25519(
    process.env.MY_PRIVATE_KEY
  );

  // If we weren't able to grab it, we should throw a new error
  if (accountIdTest == null || accountKeyTest == null) {
    throw new Error(
      "Environment variables myAccountId and myPrivateKey must be present"
    );
  }

  const client = Client.forTestnet();
  client.setOperator(accountIdTest, accountKeyTest);

  //Get the contract bytecode
  const bytecode = htsContract.data.bytecode.object;

  //Treasury Key
  const treasuryKey = PrivateKey.generateED25519();

  //Create token treasury account
  const treasuryAccount = new AccountCreateTransaction()
    .setKey(treasuryKey)
    .setInitialBalance(new Hbar(100))
    .setAccountMemo("treasury account");

  //Submit the transaction to a Hedera network
  const submitAccountCreateTx = await treasuryAccount.execute(client);

  //Get the receipt of the transaction
  const newAccountReceipt = await submitAccountCreateTx.getReceipt(client);

  //Get the account ID from the receipt
  const treasuryAccountId = newAccountReceipt.accountId;

  console.log("The new account ID is " + treasuryAccountId);

  //Create a token to interact with
  const createToken = new TokenCreateTransaction()
    .setTokenName("HTS demo")
    .setTokenSymbol("H")
    .setTokenType(TokenType.FungibleCommon)
    .setTreasuryAccountId(treasuryAccountId)
    .setInitialSupply(500);

  //Sign with the treasury key
  const signTokenTx = await createToken.freezeWith(client).sign(treasuryKey);

  //Submit the transaction to a Hedera network
  const submitTokenTx = await signTokenTx.execute(client);

  //Get the token ID from the receipt
  const tokenId = await (await submitTokenTx.getReceipt(client)).tokenId;

  //Log the token ID
  console.log("The new token ID is " + tokenId);

  //Create a file on Hedera and store the hex-encoded bytecode
  const fileCreateTx = new FileCreateTransaction().setContents(bytecode);

  //Submit the file to the Hedera test network signing with the transaction fee payer key specified with the client
  const submitTx = await fileCreateTx.execute(client);

  //Get the receipt of the file create transaction
  const fileReceipt = await submitTx.getReceipt(client);

  //Get the file ID from the receipt
  const bytecodeFileId = fileReceipt.fileId;

  //Log the file ID
  console.log("The smart contract byte code file ID is " + bytecodeFileId);

  //Deploy the contract instance
  const contractTx = await new ContractCreateTransaction()
    //The bytecode file ID
    .setBytecodeFileId(bytecodeFileId)
    //The max gas to reserve
    .setGas(2000000);

  //Submit the transaction to the Hedera test network
  const contractResponse = await contractTx.execute(client);

  //Get the receipt of the file create transaction
  const contractReceipt = await contractResponse.getReceipt(client);

  //Get the smart contract ID
  const newContractId = contractReceipt.contractId;

  //Log the smart contract ID
  console.log("The smart contract ID is " + newContractId);

  //Associate the token to an account using the SDK
  const transaction = new TokenAssociateTransaction()
    .setAccountId(accountIdTest)
    .setTokenIds([tokenId])
    .freezeWith(client);

  //Sign the transaction with the client
  const signTx = await transaction.sign(accountKeyTest);

  //Submit the transaction
  const submitAssociateTx = await signTx.execute(client);

  //Get the receipt
  const txReceipt = await submitAssociateTx.getReceipt(client);

  //Get transaction status
  const txStatus = txReceipt.status;

  console.log("The associate transaction was " + txStatus.toString());

  //Approve the token allowance
  const transactionAllowance = new AccountAllowanceApproveTransaction()
    .approveTokenAllowance(tokenId, treasuryAccountId, newContractId, 100)
    .freezeWith(client);

  //Sign the transaction with the owner account key
  const signTxAllowance = await transactionAllowance.sign(treasuryKey);

  //Sign the transaction with the client operator private key and submit to a Hedera network
  const txResponseAllowance = await signTxAllowance.execute(client);

  //Request the receipt of the transaction
  const receiptAllowance = await txResponseAllowance.getReceipt(client);

  //Get the transaction consensus status
  const transactionStatusAllowance = receiptAllowance.status;

  console.log(
    "The transaction consensus status for the allowance function is " +
      transactionStatusAllowance.toString()
  );

  //Transfer the new token to the account
  //Contract function params need to be in the order of the parameters provided in the tokenTransfer contract function
  const tokenTransfer = new ContractExecuteTransaction()
    .setContractId(newContractId)
    .setGas(2000000)
    .setFunction(
      "tokenTransfer",
      new ContractFunctionParameters()
        //The ID of the token
        .addAddress(tokenId.toSolidityAddress())
        //The account to transfer the tokens from
        .addAddress(treasuryAccountId.toSolidityAddress())
        //The account to transfer the tokens to
        .addAddress(accountIdTest.toSolidityAddress())
        //The number of tokens to transfer
        .addInt64(100)
    );

  //Sign the token transfer transaction with the treasury account to authorize the transfer and submit
  const signTokenTransfer = await tokenTransfer
    .freezeWith(client)
    .sign(treasuryKey);

  //Submit transfer transaction
  const submitTransfer = await signTokenTransfer.execute(client);

  //Get transaction status
  const transferTxStatus = await (
    await submitTransfer.getReceipt(client)
  ).status;

  //Get the transaction status
  console.log("The transfer transaction status " + transferTxStatus.toString());

  //Verify your account received the 100 tokens
  const newAccountBalance = new AccountBalanceQuery()
    .setAccountId(accountIdTest)
    .execute(client);

  console.log(
    "My new account balance is " + (await newAccountBalance).tokens.toString()
  );
}

void htsContractFunction();
```

**Comment by theekrystallee** [comment]
The [tutorial](https://docs.hedera.com/hedera/tutorials/smart-contracts/deploy-a-contract-using-the-hedera-token-service) has been updated.

**Pull request closed without merging**